### ISSUE TITLE
[Unet/tensorflow2] loss function with softmax output

### ISSUE THREAD

**Issue by pcasl** [NVIDIA/DeepLearningExamples#468]
Related to **UNet_Medical/TensorFlow2** 
TensorFlow2/Segmentation/UNet_Medical

**Describe the bug**
In the U-Net model, the last line, the output is with keras soft_max:
tf.keras.activations.softmax(out, axis=-1)

In the loss function, line 35, the crossentropy is calculated with tf.nn.softmax_cross_entropy_with_logits.

However, as mentioned by the tensorflow2 library, https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits
WARNING: This op expects unscaled logits, since it performs a softmax on logits internally for efficiency. **Do not call this op with the output of softmax**, as it will produce incorrect results.

So, I think this is a bug. Please correct me if I am wrong. Thank you.
 
**Expected behavior**
If it is bug, it should be change to be corrected

**Comment by mmarcinkiewicz** [comment]
Hi, it seems you are right. I'll fix the issue and run the numbers to see whether the convergence is impacted. Nice catch!
